packages/scrapers-v2/src/urls/scrape_urls.ts

```diff
@@ -0,0 +1,162 @@
+import { appendPath, loadHTML } from "../utils";
+import { CatalogHierarchy, College, CatalogPath } from "./types";
+
+/**
+ * Scrapes all catalog entries underneath the colleges.
+ * 
+ * @param start starting year (must be end year - 1)
+ * @param end   ending year
+ * @returns     a hierarchy of catalog entry links
+ */
+export const scrapeMajorLinks = async (start: number, end: number) => {
+  if (start !== end - 1) {
+    throw new Error("start should == end-1");
+  }
```

Comment:
wait am i tripping or do we like not need end year?

Reply:
oh I think it's like if we want to scrape a specific year, then like people shouldn't put something like 2020-2023

Reply:
but this part is subject to change!

Reply:
Seeing as in the lexer portion of the scraper, we are categorising majors using the first year, so 2022 for 2022-2023, I think it makes sense to keep it consistent and have it be "2022" referring to catalog year 2022-2023 for consistency.